agents/types/utils.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/synapsecns/sanguine/core"
+	"github.com/synapsecns/sanguine/ethergo/signer/signer"
+)
+
+// SignEncoder encodes a type, and signs it with the given salt.
+func signEncoder(ctx context.Context, signer signer.Signer, encoder Encoder, salt string) (signer.Signature, []byte, common.Hash, error) {
+	// Encode the given type.
+	encoded, err := encoder.Encode()
+	if err != nil {
+		return nil, nil, common.Hash{}, fmt.Errorf("could not encode receipt: %w", err)
+	}
+
+	// Hash the encoded type, and concatenate with hashed salt.
+	hashedEncoded := crypto.Keccak256Hash(encoded).Bytes()
+	toSign := append(crypto.Keccak256Hash([]byte(salt)).Bytes(), hashedEncoded...)
+	hashedDigest, err := HashRawBytes(toSign)
+	if err != nil {
+		return nil, nil, common.Hash{}, fmt.Errorf("could not hash receipt: %w", err)
+	}
+
+	// Sign the message.
+	signature, err := signer.SignMessage(ctx, core.BytesToSlice(hashedDigest), false)
```

Comment:
is there ever a scenario where we would want to sign a payload hash without the `\x19Ethereum...`?

Reply:
ah actually i dont think so. ignore.

Reply:
Not that I know of! If we need to customize this can always extend this function or add a new one